fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -4,10 +4,86 @@
 # LICENSE file in the root directory of this source tree.
 
 from enum import Enum
+from typing import Optional
 
+import torch
 from torch import Tensor
 
 
+# Helper Functions
+def _get_k_for_topk(topk_percent: Optional[float], top_k_element: Optional[int], top_k_total_size: int) -> int:
+    """Converts the top_k_percent to top_k_element when top_k_percent is provided
+    as the criterion for top-k calculation. When, top_k_element is used as the criterion,
+    simply returns the value for k. Also, ensures k is never 0 to avoid all sparse tensors.
+    """
+    if top_k_element is None:
+        top_k_element = int(top_k_total_size * topk_percent / 100.0)
+
+    # ensure we never have 100% sparsity in tensor and always have 1 surviving element!
+    top_k_element = max(1, top_k_element)
+    return top_k_element
+
+
+def _scatter_topk_to_sparse_tensor(
+    top_k_tensor: Tensor, to_be_sparsify_tensor: Tensor, k: int, dim: Optional[int]
+) -> Tensor:
+    """Scatter the topk values of the to_be_sparsify_tensor to a zero tensor of the same shape
+    at the top-k indices of the top_k_tensor. This function allows top-k computation with a
+    derived tensor from to_be_sparsify_tensor.
+
+    Args:
+        top_k_tensor (Tensor)
+            The source tensor whose top-k "indices" are taken and used to extract
+            the corresponding "values" from the to_be_sparsify_tensor.
+        to_be_sparsify_tensor (Tensor)
+            The tensor whose values are gathered according to the top-k indices
+            of the top_k_tensor, and a zero tensor of same shape is populated with these
+            values at those indices and creates the sparse_tensor tensor.
+        k (int)
+            the value of k for top-k
+        dim (int, optional)
```

Comment:
this is not optional. I think this is easy to confuse us.

There is an arg being optional, which means it has a default value.

Type hint can be "Optional" with a big O, that means the type is either None or the type inside Optional[]. 

Here, type is Optional, but arg is not optional. I got confused initially too.

Reply:
I suggest change to:

`dim (Optional[int]):`